### PR TITLE
First pass through NsOneProvider

### PR DESCRIPTION
**Pull request by ross**
This is a quick pass through adding basic record support for NS1. 

* [x] Tests

### CODE REVIEW

File: octodns/provider/nsone.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+from logging import getLogger
+from nsone import NSONE
+from nsone.rest.errors import ResourceException
+
+from ..record import Record
+from .base import BaseProvider
+
+
+class NsOneProvider(BaseProvider):
```
Comment (weyrick) [inline octodns/provider/nsone.py]: despite our python client still using our older `nsone` moniker, we were hoping you could shift this to be `NS1Provider` and in general use `ns1` instead of `nsone`, which is what we've rebranded to since the original python client came out.

File: octodns/provider/nsone.py [hunk 2]
```diff
@@ -0,0 +1,189 @@
...
+    '''
+    NsOne provider
+
+    nsone:
+        class: octodns.provider.nsone.NsOneProvider
+        api_key: env/NS_ONE_API_KEY
+    '''
+    SUPPORTS_GEO = False
```
Comment (weyrick) [inline octodns/provider/nsone.py]: it seems like it may be fairly easy to get geo support in, let us know if we can help with that!

File: octodns/provider/nsone.py (hunk 1 above)
Comment (ross) [inline octodns/provider/nsone.py]: Done.

File: octodns/provider/nsone.py (hunk 2 above)
Comment (ross) [inline octodns/provider/nsone.py]: Some have been easier than others. Haven't looked deeply enough at NS1's model & api to know what it'd involve. Don't plan to add it in this round since I won't have enough cycles to dig in. Once this is merged I'd be happy to look over a PR that adds support for it 😁.

**Comment by ross** [comment]
@weyrick tests have been fleshed out and a few fixes put in place. If you want to 👀  it again and give a 👍  / 👎  I think it's ready for it.

**Comment by weyrick** [comment]
👍  thanks!

**Comment by williamsjj** [comment]
What level of geo support is included in this version of the NS1 provider?

**Comment by ross** [comment]
> What level of geo support is included in this version of the NS1 provider?

There's currently no geo support in `Ns1Provider`. This PR was just a first pass to get basic record support in place. Working with the geo dns apis of providers is usually quite a bit more involved and testing all of the edge cases takes a bit of work as well. That's not something I'll be able to tackle in the near-term as since its more time that I can spare for the purpose in nights & weekends.

**Comment by williamsjj** [comment]
@ross That's too bad but I totally understand.

@weyrick Any chance NS1 can contribute geo support? Understanding it will be a limited application of filter chains to achieve that. NS1 keeps banging on us to go multi-DNS everytime there's an outage, so this would help that a lot.

**Comment by vanbroup** [comment]
See also ns1/nsone-python#28
